app/stac_api/models.py

```diff
@@ -310,12 +331,10 @@ def update_properties_eo_gsd(self, asset_eo_gsd):
             self.save()
 
     def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
-        # TODO: check if collection's bbox needs to be updated
-        # --> this could probably best be done with GeoDjango? (@Tobias)
-        # I leave this open for the moment.
-        self.collection.update_extent(self.properties_datetime)
+        self.collection.update_temporal_extent(self.properties_datetime)
 
         super().save(force_insert, force_update, using, update_fields)
+        self.collection.update_spatial_bbox(self.geometry)
```

Comment:
I would call the update extent functions only if needed, meaning only if the `properties_datetime` and/or `geometry` changed.